### ISSUE TITLE
AWS SageMakerの`create-training-job --hyper-parameters`値が保護されない

### ISSUE THREAD

**Issue by jyup-escape** [EdamAme-x/pentect#782]
# SEC-342 AWS SageMakerの`create-training-job --hyper-parameters`値が保護されない

## 概要

AWS CLIの`aws sagemaker create-training-job`は、学習アルゴリズムへハイパーパラメーターを渡せます。AWS公式資料では、`--hyper-parameters`をキーと値のマップとして定義し、JSON構文と短縮構文を記載しています。

この入力へ、一般的な秘密語を含まない任意名の合成キーと値を置いたところ、値が平文のまま出力され、診断は`masked 0`となりました。AWS公式資料は、ハイパーパラメーターへアカウント識別子、秘密、トークンなどを含めないよう注意しています。

## 影響範囲

- `aws sagemaker create-training-job --hyper-parameters`
- JSON構文および短縮構文のキーと値
- SageMaker学習ジョブ、機械学習自動化、継続的インテグレーション
- 学習コンテナーへ渡す接続情報、外部サービス認証値、署名用秘密
- 端末記録、シェル履歴、学習ジョブ起動ログ、継続的インテグレーション記録、AIエージェント入力

ハイパーパラメーターはアルゴリズム固有の設定値として扱われますが、実際の学習コードが外部サービスへの接続情報や処理用の認証値を受け取る用途にも利用できます。キー名が`PASSWORD`や`TOKEN`を含まない場合でも、値本体を保護する必要があります。

## 再現手順

実在のAWSアカウント、SageMaker学習ジョブ、IAM認証情報、外部通信は使っていません。合成値を含むAWS CLI文字列を、作業ツリーで再構築したPentect実行ファイルの`mask`処理へ標準入力から渡しました。AWS CLI、SageMaker API、学習ジョブ開始は実行していません。

### JSON構文で任意名の複数値を指定する場合

```powershell
$exe = 'D:\pentect-target-bugbounty-20260825\debug\pentect.exe'
$input = 'aws sagemaker create-training-job --training-job-name synthetic-training --role-arn arn:aws:iam::000000000000:role/synthetic-role --algorithm-specification {"TrainingImage":"example.invalid/sagemaker:latest","TrainingInputMode":"File"} --output-data-config {"S3OutputPath":"s3://synthetic-bucket/output"} --resource-config {"InstanceType":"ml.t3.medium","InstanceCount":1,"VolumeSizeInGB":1} --stopping-condition {"MaxRuntimeInSeconds":60} --hyper-parameters {"FIRST":"SageMakerHyperFirst987!","SECOND":"SageMakerHyperSecond987!"}'
$input | & $exe mask --kind text
```

実際の出力:

```text
aws sagemaker create-training-job --training-job-name synthetic-training --role-arn arn:aws:iam::000000000000:role/synthetic-role --algorithm-specification {"TrainingImage":"example.invalid/sagemaker:latest","TrainingInputMode":"File"} --output-data-config {"S3OutputPath":"s3://synthetic-bucket/output"} --resource-config {"InstanceType":"ml.t3.medium","InstanceCount":1,"VolumeSizeInGB":1} --stopping-condition {"MaxRuntimeInSeconds":60} --hyper-parameters {"FIRST":"SageMakerHyperFirst987!","SECOND":"SageMakerHyperSecond987!"}
[pentect] profile=Strict masked 0 value(s), 0 warned.
```

任意名の二つのハイパーパラメーター値が平文のまま出力されました。

### 短縮構文で複数値を指定する場合

```text
aws sagemaker create-training-job --training-job-name synthetic-training --role-arn arn:aws:iam::000000000000:role/synthetic-role --algorithm-specification {"TrainingImage":"example.invalid/sagemaker:latest","TrainingInputMode":"File"} --output-data-config {"S3OutputPath":"s3://synthetic-bucket/output"} --resource-config {"InstanceType":"ml.t3.medium","InstanceCount":1,"VolumeSizeInGB":1} --stopping-condition {"MaxRuntimeInSeconds":60} --hyper-parameters DB_URL=SageMakerHyperShortDb987!,FIRST=SageMakerHyperShortFirst987!
[pentect] profile=Strict masked 1 value(s), 0 warned.
```

短縮構文では一般的な検出が値の一部へ反応しましたが、マップの各値をハイパーパラメーター構造として保護した結果ではありません。JSON構文の任意名入力では両方が平文で残りました。

### 一般的な秘密語を含むキーとの比較

```text
aws sagemaker create-training-job --training-job-name synthetic-training --role-arn arn:aws:iam::000000000000:role/synthetic-role --algorithm-specification {"TrainingImage":"example.invalid/sagemaker:latest","TrainingInputMode":"File"} --output-data-config {"S3OutputPath":"s3://synthetic-bucket/output"} --resource-config {"InstanceType":"ml.t3.medium","InstanceCount":1,"VolumeSizeInGB":1} --stopping-condition {"MaxRuntimeInSeconds":60} --hyper-parameters {"DB_PASSWORD":"SageMakerHyperDb987!"}
[pentect] profile=Strict masked 1 value(s), 0 warned.
```

キー名に一般的な秘密語を含めると別の検出が反応します。この結果は、任意名のハイパーパラメーターをSageMakerの構造として保護できていない状態を示します。

### 期待される変換例

```text
aws sagemaker create-training-job --training-job-name synthetic-training --role-arn arn:aws:iam::000000000000:role/synthetic-role --algorithm-specification {"TrainingImage":"example.invalid/sagemaker:latest","TrainingInputMode":"File"} --output-data-config {"S3OutputPath":"s3://synthetic-bucket/output"} --resource-config {"InstanceType":"ml.t3.medium","InstanceCount":1,"VolumeSizeInGB":1} --stopping-condition {"MaxRuntimeInSeconds":60} --hyper-parameters {"FIRST":"PROTECTED_HANDLE","SECOND":"PROTECTED_HANDLE"}
[pentect] profile=Strict masked 2 value(s), 0 warned.
```

保護ハンドルは説明用の表記です。実行結果から取得したハンドルや実在の秘密値を報告本文へ保存していません。

## 公式仕様との対応

[AWS公式の`aws sagemaker create-training-job`資料](https://docs.aws.amazon.com/cli/latest/reference/sagemaker/create-training-job.html)は、`--hyper-parameters`を学習アルゴリズムへ渡すキーと値のマップとして定義しています。同資料はJSON構文と`KeyName1=string,KeyName2=string`形式を掲載しています。

同資料は、ハイパーパラメーターへアカウント識別子、秘密、トークンなどの機密情報を含めないよう注意しています。今回の入力は、公式の学習ジョブ構造へ合成ハイパーパラメーターを配置したものです。

## 期待される動作

SageMakerのハイパーパラメーター指定を認識し、キー名によらず値本体を保護することが期待されます。

- JSON構文の単一および複数のキーと値
- 短縮構文の単一および複数のキーと値
- 空白、記号、接続文字列形式の値
- `--cli-input-json`内の`HyperParameters`値
- 一般的な秘密語を含まないキー名の値

学習ジョブ名、アルゴリズム設定、キー名、ハイパーパラメーター値を分けて扱い、値本体が端末出力や後続の外部処理へ平文で渡らないことが求められます。

## 確認状況

- 使用した値: `SageMakerHyperFirst987!`、`SageMakerHyperSecond987!`、`SageMakerHyperShortDb987!`、`SageMakerHyperShortFirst987!`
- 実在の認証情報: 使用していません
- AWSへの接続: 実施していません
- Pentect実行結果: 任意名JSON入力で`masked 0`、`warned 0`
- 公開本文へ記載した値: すべて合成値
- 期待される保護後表記: `PROTECTED_HANDLE`
